plugins/modules/ini_file.py

```diff
@@ -305,14 +379,22 @@ def do_ini(module, filename, section=None, option=None, values=None,
     section_lines = []
 
     for index, line in enumerate(ini_lines):
+        # end of section:
+        if within_section and line.startswith(u'['):
```

Comment:
All code in Ansible is unicode by definition. This is redundant:
```suggestion
        if within_section and line.startswith('['):
```
Same applies to other occurrences - yes, I see the existing code makes use of that. We should review all occurrences in another PR.

Reply:
Actually we should NOT do that while we still support Python 2.7.

Reply:
In my memory, `unicode` was in the mandatory imports from `__future__`. That clearly not being the case, I take it back.